Intersect.Server.Framework/Items/IItemSource.cs

```diff
@@ -0,0 +1,11 @@
+using Intersect.Enums;
+using Intersect.Server.Framework.Entities;
+
+namespace Intersect.Server.Framework;
+
+public interface IItemSource
+{
+    Guid Id { get; }
+    EntityType EntityType { get; }
```

Comment:
I made a mistake in my original description in Discord, `EntityType` should be in `EntityItemSource` and this enum should be added and referenced here as `ItemSourceType SourceType { get; }` which `EntityItemSource` should implement as `public ItemSourceType SourceType => ItemSourceType.Entity;`:

```csharp
public enum ItemSourceType
{
    Unknown = 0,
    Entity = 1,
}
```

With respect to my comment from another file, I honestly want another enum value `Map = 2`.